Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -0,0 +1,714 @@
+        registerClass(TopTabsBackgroundDecorationView.self, forDecorationViewOfKind: TopTabsBackgroundDecorationView.Identifier)
+    }
+    
+    override func shouldInvalidateLayoutForBoundsChange(newBounds: CGRect) -> Bool {
+        return true
+    }
+    
+    // MARK: layoutAttributesForElementsInRect
+    override func layoutAttributesForElementsInRect(rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        var attributes = super.layoutAttributesForElementsInRect(rect)!
+        
+        // Create decoration attributes
+        let decorationAttributes = UICollectionViewLayoutAttributes(forDecorationViewOfKind: TopTabsBackgroundDecorationView.Identifier, withIndexPath: NSIndexPath(forRow: 0, inSection: 0))
+        
+        // Make the decoration view span the entire row
+        let size = collectionViewContentSize()
+        decorationAttributes.frame = CGRectMake(-(TopTabsUX.TopTabsBackgroundPadding-TopTabsUX.TopTabsBackgroundShadowWidth*2)/2, 0, size.width+(TopTabsUX.TopTabsBackgroundPadding-TopTabsUX.TopTabsBackgroundShadowWidth*2), size.height)
+        
+        // Set the zIndex to be behind the item
+        decorationAttributes.zIndex = -1
+        
+        // Add the attribute to the list
+        attributes.append(decorationAttributes)
+        
+        return attributes
+    }
+    
+    override func layoutAttributesForDecorationViewOfKind(elementKind: String, atIndexPath indexPath: NSIndexPath) -> UICollectionViewLayoutAttributes? {
+        return super.layoutAttributesForDecorationViewOfKind(elementKind, atIndexPath: indexPath)
+    }
+}
+
+private class TopTabFader: UIView {
+    lazy var hMaskLayer: CAGradientLayer = {
+        let innerColor: CGColorRef = UIColor(white: 1.0, alpha: 1.0).CGColor
+        let outerColor: CGColorRef = UIColor(white: 1.0, alpha: 0.0).CGColor
+        let hMaskLayer = CAGradientLayer()
+        hMaskLayer.colors = [outerColor, innerColor, innerColor, outerColor]
+        hMaskLayer.locations = [0.00, 0.03, 0.97, 1.0]
+        hMaskLayer.startPoint = CGPointMake(0, 0.5)
+        hMaskLayer.endPoint = CGPointMake(1.0, 0.5)
+        hMaskLayer.anchorPoint = CGPointZero
+        return hMaskLayer
+    }()
+    
+    init() {
+        super.init(frame: CGRectZero)
+        layer.mask = hMaskLayer
+    }
+    
+    private override func layoutSubviews() {
+        super.layoutSubviews()
+        hMaskLayer.locations = [0.00, 15/frame.width, 1-15/frame.width, 1.0]
+        hMaskLayer.frame = CGRectMake(0, 0, frame.width, frame.height)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class TopTabsBackgroundDecorationView : UICollectionReusableView {
+    static let Identifier = "TopTabsBackgroundDecorationViewIdentifier"
+    private lazy var right = SingleCurveView(right: true)
+    private lazy var left = SingleCurveView(right: false)
```

Comment:
Nit: I would add `Curve` to these variables just to add some more clarity like `rightCurve` and `leftCurve`.